service-commands/src/commands/users.js

```diff
@@ -22,23 +25,55 @@ User.uploadProfileImagesAndAddUser = async (libsWrapper, metadata, userPicturePa
   await waitForIndexing()
   metadata = await User.getUser(libsWrapper, userId)
 
-  // Upload images to that primary (will inherently sync)
-  const userPicFile = fs.createReadStream(userPicturePath)
+  // Upload photo for profile picture
+  await User.uploadPhotoAndUpdateMetadata({
+    metadata,
+    libsWrapper,
+    userId,
+    picturePath: userPicturePath
+  })
+
+  return userId
+}
+
+/**
+ * Upload photo for cover photo and profile picture and update the metadata object
+ * @param {Object} param
+ * @param {Object} param.metadata original metadata object
+ * @param {Object} param.libsWrapper libs wrapper in ServiceCommands
+ * @param {number} param.userId
+ * @param {string} param.picturePath path of picture to upload
+ * @param {boolean} param.[updateCoverPhoto=true] flag to update cover_photo_sizes hash
+ * @param {boolean} param.[updateProfilePicture=true] flag to update profile_picture_sizes hash
+ */
+User.uploadPhotoAndUpdateMetadata = async ({
+  metadata,
```

Comment:
this is passed by reference and modified, pls fix and avoid in future

Reply:
I think this is fine? I don't think it's necessary to create a copy of the metadata just to potentially update those 2 fields. also, this metadata blob input is not used anywhere else https://github.com/AudiusProject/audius-protocol/blob/vg-add-replica-set-on-signup-umd/mad-dog/src/tests/test_integration.js#L385-L396

Reply:
yeah i see that this isn't used but this is still bad practice. i've made this same mistake a bunch, but we should always be passing a copy and assigning the return value if necessary. this function returns the argument metadata it should actually return a new object that is then assigned by the caller if necessary

Reply:
discussed offline -- will make a copy